src/server.jl

```diff
@@ -206,11 +201,22 @@ function serve(; ipaddr::Union{String, IPAddr}="localhost", port::Int=8000)
         end
     catch err
         if isa(err, InterruptException)
-            # NOTE ideally here we would also want to stop the listener. However this is
-            # not as easy as stopping the file watching tasks.
-            stop_tasks!(filewatcher)
+            println("\n⋮ waiting for everything to shut down")
+
+            # stop the file watcher
+            stop(filewatcher)
+
+            # close all websockets
+            for wss ∈ values(WS_HTML_FILES)
+                foreach(wsi -> close(wsi.io), wss)
+            end
             empty!(WS_HTML_FILES)
-        println("\n✓ LiveServer shut down.")
+
+            # shut down server
+            close(server)
+            while (server.status != 6) sleep(0.05) end
```

Comment:
Maybe check but I think this is instantaneous actually. So I don't think you ever enter that loop.

Reply:
Yes it is, removing that line. Came from the original version where I also checked for the (three) async tasks to return...